py_trees_parser/parser.py

```diff
@@ -404,6 +404,29 @@ def _sub_args(self, args, var):
                 self.logger.error(f"Argument '{var_name}' not found in arg list: {args}")
                 raise ValueError(f"Argument '{var_name}' not found in arg list")
 
+        # Support for argument substitution within strings
+        elif isinstance(var, str) and "${" in var and "}" in var:
+            result = var
+            # Find all ${...} patterns in the string
+            start_idx = 0
+            while "${" in result[start_idx:]:
```

Comment:
I think this will actually catch `my_${arg1}_${arg2}`. It might be good to test that.